### PR TITLE
Check if PayPal Supports App Switch

### PR DESCRIPTION
**Pull request by anibalb2500**
Thank you for your contribution to PayPal.

> Before submitting this PR, note that we cannot accept language translation PRs. We have a dedicated localization team to provide the translations. If there is an error in a specific translation, you may open an issue and we will escalate it to the localization team.

### Summary of changes

- Restrict PayPal app switch to installs that meet a minimum supported app version. `DeviceInspector.canResolvePayPalAppSwitch()` now checks the installed PayPal app's version in addition to its existing "does this URI resolve to the PayPal app" check, and fails closed (no app switch) if the version can't be determined.
- Add `DeviceInspector.payPalAppVersionName`, which reads the installed PayPal app's `PackageInfo.versionName` via `PackageManager`, returning `null` if the app isn't installed or the version can't be read.
- Add a private `DeviceInspector.isAppSwitchSupportedVersion()` helper that parses the leading major-version integer from `versionName` (e.g. `"9.1.0"` → `9`) and compares it against a new `APP_SWITCH_MIN_MAJOR_VERSION` constant.
- Update `PayPalWebCheckoutClient.canAttemptPayPalAppSwitch()` KDoc to reference the new version-gating behavior now living in `DeviceInspector`. No logic change in this class — the check is fully encapsulated in `DeviceInspector.canResolvePayPalAppSwitch()`, so this only affects the shopper-session (v3) app-switch flow, not the legacy `PatchCCOWithAppSwitchEligibility` path.
- Add unit test coverage in `DeviceInspectorUnitTest` for the new version accessor and gating logic: exact boundary, below-minimum, undeterminable, and unparseable version cases.

Resolves [DTPPMOBILE-547](https://paypal.atlassian.net/browse/DTPPMOBILE-547).

### Checklist

- [ ] Added a changelog entry

### Authors

> List GitHub usernames for everyone who contributed to this pull request.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
   * Add `PayPalWebCheckoutClient.vault(context, setupTokenId, callback)` method (v3) — requires a prior call to `createPayPalSession()`
   * Deprecate `PayPalWebCheckoutClient.start(activity, request, callback)` — use `createPayPalSession()` followed by `start(context, orderId, callback)` instead
   * Deprecate `PayPalWebCheckoutClient.vault(activity, request, callback)` — use `createPayPalSession()` followed by `vault(context, setupTokenId, callback)` instead
+  * Fix: app switch (via `createPayPalSession()` / `start()` / `vault()`) is now only attempted when the installed PayPal app meets a minimum supported version; older, incompatible versions fall back to Chrome Custom Tabs checkout
```
Comment (kgangineni) [inline CHANGELOG.md]: whole PR is internal functionality changelog entry is not needed

**Review by kgangineni — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (anibalb2500) [reply CHANGELOG.md]: Removed
